src/main/java/org/opensearch/knn/training/TrainingDataConsumer.java

```diff
@@ -34,13 +45,29 @@ public TrainingDataConsumer(NativeMemoryAllocation.TrainingDataAllocation traini
         this.trainingDataAllocation = trainingDataAllocation;
     }
 
-    @Override
-    public void accept(List<Float[]> floats) {
-        trainingDataAllocation.setMemoryAddress(
-            JNIService.transferVectors(
-                trainingDataAllocation.getMemoryAddress(),
-                floats.stream().map(ArrayUtils::toPrimitive).toArray(float[][]::new)
-            )
-        );
+    protected abstract void accept(List<T> vectors);
```

Comment:
We can remove accept, extractVectorsFromHits, getTotalAddedVectors which is not being called outside of this class.

Reply:
Removed extractVectorsFromHits, getTotalAddedVectors and kept accept to expose for some tests